source/Octopus.Tentacle/Kubernetes/IKubernetesPodLogService.cs

```diff
@@ -21,39 +22,65 @@ class KubernetesPodLogService : KubernetesService, IKubernetesPodLogService
         readonly IKubernetesPodMonitor podMonitor;
         readonly ITentacleScriptLogProvider scriptLogProvider;
         readonly IScriptPodSinceTimeStore scriptPodSinceTimeStore;
+        readonly IKubernetesEventService eventService;
 
-        public KubernetesPodLogService(IKubernetesClientConfigProvider configProvider, IKubernetesPodMonitor podMonitor, ITentacleScriptLogProvider scriptLogProvider, IScriptPodSinceTimeStore scriptPodSinceTimeStore, ISystemLog log) 
+        public KubernetesPodLogService(
+            IKubernetesClientConfigProvider configProvider,
+            IKubernetesPodMonitor podMonitor,
+            ITentacleScriptLogProvider scriptLogProvider,
+            IScriptPodSinceTimeStore scriptPodSinceTimeStore,
+            IKubernetesEventService eventService,
+            ISystemLog log)
             : base(configProvider, log)
         {
             this.podMonitor = podMonitor;
             this.scriptLogProvider = scriptLogProvider;
             this.scriptPodSinceTimeStore = scriptPodSinceTimeStore;
+            this.eventService = eventService;
         }
 
         public async Task<(IReadOnlyCollection<ProcessOutput> Outputs, long NextSequenceNumber)> GetLogs(ScriptTicket scriptTicket, long lastLogSequence, CancellationToken cancellationToken)
         {
             var tentacleScriptLog = scriptLogProvider.GetOrCreate(scriptTicket);
             var podName = scriptTicket.ToKubernetesScriptPodName();
 
-            var podLogs = await GetPodLogs();
+            var podLogsTask = GetPodLogs();
+            var podEventsTask = GetPodEvents(scriptTicket, podName, cancellationToken);
+            await Task.WhenAll(podLogsTask, podEventsTask);
```

Comment:
Is there any reason to await both tasks here when we await both separately below?

Reply:
the awaits below won't actually re-run the code, it's just a quick way of getting the result (without calling `.Result)

Reply:
I meant that we don't need the `await Task.WhenAll(podLogsTask, podEventsTask);`
It's nicer to only await the tasks when we process them IMO, since we'd only be waiting for the data we're actively using.

Reply:
ah, I see what you mean. Yeah. I can see it.